File: privacystreams-app/src/main/java/io/github/privacystreams/app/Controllers.java
package io.github.privacystreams.app;


import android.content.Context;
import android.content.Intent;

import io.github.privacystreams.core.actions.collect.Collectors;

public class Controllers {
    private Context context;

    Controllers(Context context) {
        this.context = context;
    }

    public void startCollectService() {
        Intent collectServiceIntent = new Intent(this.context, PStreamCollectService.class);
        this.context.startService(collectServiceIntent);
    }

    public void stopCollectService() {
        Intent collectServiceIntent = new Intent(this.context, PStreamCollectService.class);
        this.context.stopService(collectServiceIntent);
    }
}

File: privacystreams-app/src/main/java/io/github/privacystreams/app/Controllers.kt
package io.github.privacystreams.app


import android.content.Context
import android.content.Intent
import io.github.privacystreams.app.db.PSGeolocationTable
import io.github.privacystreams.app.db.PSNotificationTable
import io.github.privacystreams.core.UQI
import io.github.privacystreams.core.purposes.Purpose

class Controllers internal constructor(private val context: Context) {

    fun startCollectService() {
        val collectServiceIntent = Intent(this.context, PStreamCollectService::class.java)
        this.context.startService(collectServiceIntent)
    }

    fun stopCollectService() {
        val collectServiceIntent = Intent(this.context, PStreamCollectService::class.java)
        this.context.stopService(collectServiceIntent)
    }

    fun test() {
        UQI(context).getData(PSGeolocationTable.PROVIDER(), Purpose.TEST("Test"))
                .limit(10)
                .debug();

        UQI(context).getData(PSNotificationTable.PROVIDER(), Purpose.TEST("Test"))
                .limit(10)
                .debug();
    }
}
